### PR TITLE
Implement history trimming with GPT4 tokenizer

### PR DESCRIPTION
**Pull request by calyptobai**
This uses [`tiktoken-rs`](https://crates.io/crates/tiktoken-rs) to obtain token counts.

### CODE REVIEW

File: server/bleep/src/webserver/answer.rs [hunk 1]
```diff
@@ -621,6 +625,40 @@ impl Conversation {
...
+        while tiktoken_rs::get_chat_completion_max_tokens("gpt-4", &tiktoken_msgs)? < HEADROOM {
+            tiktoken_msgs
+                .iter_mut()
+                .find(|m| m.role == "user" && m.content != "[HIDDEN]")
+                .context("could not find message to trim")?
+                .content = "[HIDDEN]".into();
+        }
+
+        dbg!(&tiktoken_msgs, old_len);
+
```
Comment (nerdypepper) [inline server/bleep/src/webserver/answer.rs:653]:
```suggestion
```

File: server/bleep/src/webserver/answer.rs [hunk 2]
```diff
@@ -621,6 +625,40 @@ impl Conversation {
...
+            .iter()
+            .map(|m| tiktoken_rs::ChatCompletionRequestMessage {
+                role: m.role.clone(),
+                content: m.content.clone(),
+                name: None,
+            })
+            .collect::<Vec<_>>();
+
+        let old_len = tiktoken_rs::get_chat_completion_max_tokens("gpt-4", &tiktoken_msgs)?;
+
```
Comment (nerdypepper) [inline server/bleep/src/webserver/answer.rs:643]:
```suggestion
```

**Review by nerdypepper — APPROVED** [review]
lgtm!

**Review by ggordonhall — APPROVED** [review]

**Comment by ggordonhall** [comment]
@calyptobai If we're using this, can we remove the GPT-2 tokenizer, and the `tokenizers` dependency?

**Pull request merged**